Add files via upload

amralaa2468 opened the pull request:
Question bank service full backend code.

amralaa2468 committed:
Add files via upload

Question bank service full backend code.

omarsaad77 commented on src/controllers/answers-controller.js:
```diff
@@ -0,0 +1,133 @@
+  }
+}
+//---------------------------------------------------------------
+async function deleteAnswerFromExistingQuestion(req, res, next) {
```
/what if this is the only correct answer?

omarsaad77 commented on src/controllers/questions-controller.js:
```diff
@@ -0,0 +1,295 @@
+import HttpError from "../models/http-error.js";
+import Question from "../models/question.js";
+import { Answer } from "../models/answer.js";
+
+async function createQuestion(req, res, next) {
```
where do you handle validation that question should have at least 2 answers and 1 correct answer

amralaa2468 committed:
Question bank service modified

Question bank service modified to work with keycloak service (since the token received from the keycloak is different from the token received from the normal authentication service)
